src/systemctl/systemctl.c

```diff
@@ -6380,72 +6394,13 @@ static int service_log_setting(int argc, char *argv[], void *userdata) {
         if (r < 0)
                 return r;
 
-        if (argc == 2) {
-                _cleanup_free_ char *value = NULL;
-
-                r = sd_bus_get_property_string(
-                                bus,
-                                dbus_name,
-                                "/org/freedesktop/LogControl1",
-                                "org.freedesktop.LogControl1",
-                                level ? "LogLevel" : "LogTarget",
-                                &error,
-                                &value);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to get log %s of service %s: %s",
-                                               level ? "level" : "target",
-                                               dbus_name, bus_error_message(&error, r));
-
-                puts(value);
-
-        } else {
-                assert(argc == 3);
-
-                r = sd_bus_set_property(
-                                bus,
-                                dbus_name,
-                                "/org/freedesktop/LogControl1",
-                                "org.freedesktop.LogControl1",
-                                level ? "LogLevel" : "LogTarget",
-                                &error,
-                                "s",
-                                argv[2]);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to set log %s of service %s to %s: %s",
-                                               level ? "level" : "target",
-                                               dbus_name, argv[2], bus_error_message(&error, r));
-        }
-
-        return 0;
-}
-
-static int log_target(int argc, char *argv[], void *userdata) {
-        sd_bus *bus;
-        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
-        int r;
-
-        r = acquire_bus(BUS_MANAGER, &bus);
-        if (r < 0)
-                return r;
-
-        if (argc == 1) {
-                _cleanup_free_ char *target = NULL;
-
-                r = bus_get_property_string(bus, bus_systemd_mgr, "LogTarget", &error, &target);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to get log target: %s", bus_error_message(&error, r));
-
-                puts(target);
-
-        } else {
-                assert(argc == 2);
-
-                r = bus_set_property(bus, bus_systemd_mgr, "LogTarget", &error, "s", argv[1]);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to set log target: %s", bus_error_message(&error, r));
-        }
+        const BusLocator bloc = {
```

Comment:
i find "const" weird for something that isn't really const. it gets us nothing and is just confusing?

maybe put this together on the fly on the invocation log_setting_internal() below

```c
return log_setting_internal(
                bus,
                &(BusLocator) {
                        .destination = …
                },
                argv[0], argv[2]);
```

Reply:
I know this is possible, but this call is already fairly big. I find it nicer to split in two. And I think `const` is fine: it's just saying that the variable will not be modified afterwards.